src/components/exercises/ChooseArticle.js

```diff
@@ -0,0 +1,131 @@
+import { addIndex, forEach, map, toUpper } from 'ramda';
+import React from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+
+import speakerImage from '../../assets/images/speaker.png';
+import { BLACK_TRANSPARENT, PRIMARY_STRONG , PRIMARY, ARTICLES } from '../../styles/colors';
+import { DEFAULT } from '../../styles/text';
+import { loadSounds } from '../helpers/audio';
+import { RoundImageWithButton } from '../Components';
+
+const mapIndexed = addIndex(map);
+
+const styles = {
+  letterStyle: {
+    backgroundColor: PRIMARY_STRONG,
+  },
+  article: {
+    marginHorizontal: 10,
+    padding: 10
+  },
+  shared: {
+    padding: 5,
+    margin: 2,
+    borderRadius: 20,
+    elevation: 10,
+    shadowColor: BLACK_TRANSPARENT,
+    shadowOpacity: 1,
+    shadowRadius: 0,
+    shadowOffset: {
+      height: 4,
+      width: 4
+    }
+  }
+};
+
+class ChooseArticle extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      selectedArticle: ''
+    }
+  }
+
+  play = () => {
+    const playAll = ([sound, ...rest]) => {
+      sound.playAsync();
+      sound.setPlaybackFinishedCallback(() => {
+        sound.setPositionAsync(0);
+        if (rest.length > 0) {
+          setTimeout(() => playAll(rest), 1000);
+        }
+      });
+    };
+    playAll(this.props.sounds);
+  };
+
+  selectArticle = key => () => {
+    this.setState(({ selectedArticle }) => {
+      if (selectedArticle === key) {
+        return { selectedArticle: '' };
+      }
+
+      return { selectedArticle: key };
+    });
+  }
+
+  render() {
+    const letters = mapIndexed(
+      (char, key) => (
+        <View
+          key={key}
+          style={[
+            { padding: 5 },
+            styles.letterStyle,
+            styles.shared
+          ]}
+        >
+          <Text style={DEFAULT}>
+            {char}
+          </Text>
+        </View>
+      ),
+      this.props.text
+    );
+
+    const articles = mapIndexed(
+      (article, key) => (
+        <TouchableOpacity
+          key={key}
+          style={[
+            styles.shared,
+            styles.article,
+            {backgroundColor: this.state.selectedArticle === article ? ARTICLES[toUpper(article)].STRONG : ARTICLES[toUpper(article)].WEAK}
```

Comment:
I would extract that into its own function since this line is rather hard to read. So more something like that:

```.js
<TouchableOpacity
  style={[
    styles.shared,
    styles.article,
    this.getArticleStyle(article)
  ]}
/>
```
and
```.js
getArticleStyle = article => {
  const backgroundColor = this.state.selectedArticle === article
    ? ARTICLES[toUpper(article)].STRONG
    : ARTICLES[toUpper(article)].WEAK

  return { backgroundColor }
}
```